internal/support/notifications/rest_subscription.go

```diff
@@ -117,40 +250,33 @@ func subscriptionByIDHandler(w http.ResponseWriter, r *http.Request) {
 	switch r.Method {
 	case http.MethodGet:
 
-		s, err := dbClient.GetSubscriptionById(id)
+		s, err := getSubscriptionByID(id)
 		if err != nil {
-			if err == db.ErrNotFound {
-				http.Error(w, "Subscription not found", http.StatusNotFound)
-			} else {
+			switch err.(type) {
+			case errors.ErrSubscriptionNotFound:
+				http.Error(w, err.Error(), http.StatusNotFound)
+			default:
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 			}
-			LoggingClient.Error(err.Error())
 			return
 		}
 
 		pkg.Encode(s, w, LoggingClient)
 		break
 	case http.MethodDelete:
-		_, err := dbClient.GetSubscriptionById(id)
-		if err != nil {
-			if err == db.ErrNotFound {
-				http.Error(w, "Subscription not found", http.StatusNotFound)
-			} else {
+
+		if err := deleteSubscriptionByID(id); err != nil {
+			switch err.(type) {
+			case errors.ErrSubscriptionNotFound:
+				http.Error(w, err.Error(), http.StatusNotFound)
+			default:
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 			}
-			LoggingClient.Error(err.Error())
-			return
-		}
-
-		LoggingClient.Info("Deleting subscription: " + id)
-
-		if err = dbClient.DeleteSubscriptionById(id); err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			LoggingClient.Error(err.Error())
 			return
 		}
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("true"))
```

Comment:
Why did you add this?

Reply:
Because I saw it was like this in core data so I added it for consistency